src/natcap/invest/coastal_vulnerability.py

```diff
@@ -1885,21 +1930,103 @@ def _schedule_habitat_tasks(
             working_dir, '%s%s.pickle' %
             (habitat_row.id, file_suffix))
         habitat_pickles_list.append(target_habitat_pickle_path)
-        habitat_task_list.append(task_graph.add_task(
-            func=search_for_habitat,
-            args=(base_shore_point_vector_path,
-                  habitat_row.distance,
-                  habitat_row.rank,
-                  habitat_row.id,
-                  base_habitat_path,
-                  target_habitat_pickle_path),
-            target_path_list=[target_habitat_pickle_path],
-            task_name='searching for %s' % habitat_row.id))
+        vector = gdal.OpenEx(base_habitat_path, gdal.OF_VECTOR)
+        if vector:
+            vector = None
+            habitat_task_list.append(task_graph.add_task(
+                func=search_for_vector_habitat,
+                args=(base_shore_point_vector_path,
+                      habitat_row.distance,
+                      habitat_row.rank,
+                      habitat_row.id,
+                      base_habitat_path,
+                      target_habitat_pickle_path),
+                target_path_list=[target_habitat_pickle_path],
+                task_name='searching for %s' % habitat_row.id))
+            continue
+        raster = gdal.OpenEx(base_habitat_path, gdal.OF_RASTER)
+        if raster:
+            raster = None

```

Comment:
For the spatial type checking (raster/vector), this sounds like a good use case for `pygeoprocessing.get_gis_type()`.  It doesn't do much more than what you have here with the conditionals, but would mean that you wouldn't need to remember to `None` out the raster/vector variables.

Reply:
Cool, I did not know that existed